### PR TITLE
Unbreak side tracing transition

### PR DESCRIPTION
**Pull request by ltratt**
The main part of this PR is in https://github.com/ykjit/yk/commit/6c02c682f5c84f7685c46254deff4824bfd84798.

Fixes https://github.com/ykjit/yk/issues/1302.

[There is at least one other thing I want to fix, but it's not directly related to #1302 so I'll do it separately.]

### CODE REVIEW

File: ykrt/src/location.rs [hunk 1]
```diff
@@ -249,15 +249,15 @@ pub(crate) enum HotLocationKind {
...
         /// deep) side trace that ultimately relates to this [CompiledTrace], other threads can
         /// execute this compiled trace.
-        ctr: Arc<dyn CompiledTrace>,
+        root_ctr: Arc<dyn CompiledTrace>,
         /// The ID of the guard that failed (inside `parent`).
         guardid: usize,
-        /// The [CompiledTrace] that the guard failed in. This will either be equal to `ctr` or a
-        /// (possibly many levels deep) child trace of `ctr`.
-        parent: Arc<dyn CompiledTrace>,
+        /// The [CompiledTrace] that the guard failed in. This will either be `ctr` or a descendent
```
Comment (ptersilie) [inline ykrt/src/location.rs:258]: "either be `root_ctr`"

**Comment by ptersilie** [comment]
Only one comment.

File: ykrt/src/location.rs (hunk 1 above)
Comment (ltratt) [reply ykrt/src/location.rs]: Fixed in 0ba829a3.

**Comment by ptersilie** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**